### PR TITLE
Fix and update code documentation in RTD

### LINKED ISSUE #278: [documentation] fix code documentation
The code documentation is not displayed, well, see [here](https://pvcompare.readthedocs.io/en/latest/code.html#).

Goal: 
- [x] one can click on the function names
- [~] first line statement of the docstrings is displayed in second column --> will be finalized in #288
- [x] check whether all functionalities do appear

### PR DESCRIPTION
**Pull request by SabineHaas**
Fix #278, 

**Changes proposed in this pull request**:
- Update code documentation in RTD: add missing functions and modules and delete outdated ones
- Minor changes in docstrings 
- Move docs requirements from `docs/docs_requirements.txt ` to `setup.py` - now installed e.g. by `pip install -e .[dev, docs]` (#276)
- Add config file for RTD `readthedocs.yml` (#276)

The following steps were realized, as well (required):
- [x] Update the CHANGELOG.md
- :x: Check if full simulation tests pass locally (`EXECUTE_TESTS_ON=master pytest`)

Also the following steps were realized (if applies):
- [ ] Use in-line comments to explain your code
- [ ] Write docstrings to your code
- [ ] For new functionalities: Explain in readthedocs
- [ ] Write test(s) for your new patch of code

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

In case of an error due to linting, run `black . --exclude docs/` and push your changes.
Note that in case you do not fix a whole issue you should start your PR with `Address #xyz`.

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/mvs_eland/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

**Comment by SabineHaas** [comment]
This PR is not yet ready for review - code documentation works locally but is not displayed well online.

**Comment by SabineHaas** [comment]
> This PR is not yet ready for review - code documentation works locally but is not displayed well online.

It's working now, see [docs](https://pvcompare.readthedocs.io/en/fix-code_documentation_rtd/code.html) :)

**Comment by SabineHaas** [comment]
@MaGering I have no idea why the RTD build is not triggered for this PR anymore. I've activated the branch so that you can see the version of this PR there, see comment above.

File: docs/code.rst [hunk 1]
```diff
@@ -92,24 +96,57 @@ Function for calculating the feed-in time series for the perovskite silicone tec
...
+
+    heat_pump_and_chiller.add_sector_coupling
+    heat_pump_and_chiller.calculate_cops_and_eers
+
+.. _thermal_storage:
+
+Stratified thermal storage
+--------------------------
+
+Functions for calculating specific parameters of a stratified thermal storage and for adding it to the energy system.
```
Comment (MaGering) [inline docs/code.rst:117]:
```suggestion
Functions for calculating thermal losses of a stratified thermal storage and for adding it to the energy system.
```

File: pvcompare/check_inputs.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def check_for_valid_country_year(country, year, static_inputs_directory):
 
 def add_local_grid_parameters(static_inputs_directory, user_inputs_mvs_directory):
     """
-    Adds grid parameters.
+    Adds grid parameters to such as electricity price or feed-in tariff to energyProviders.csv.
```
Comment (MaGering) [inline pvcompare/check_inputs.py:224]:
```suggestion
    Adds grid parameters such as electricity price or feed-in tariff to energyProviders.csv.
```

File: pvcompare/plots.py [hunk 3]
```diff
@@ -255,8 +256,8 @@ def compare_weather_years(
     user_inputs_mvs_directory=None,
 ):
     """
-    Barplot that shows yearly aggregated weather parameters: ghi, dni, dhi and
-    temperature.
+    Barplot that shows yearly aggregated weather parameters: ghi, dni, dhi and temperature.
```
Comment (MaGering) [inline pvcompare/plots.py:259]:
```suggestion
    Bar plot that shows yearly aggregated weather parameters: ghi, dni, dhi and temperature.
```

**Review by MaGering — APPROVED** [review]
So nice that `Code documentation` works now! Thank you for the changes and updating everything. I just added some minor corrections. 

The docs build has been successfully triggered now, hasn't it?

**Comment by SabineHaas** [comment]
> So nice that `Code documentation` works now! Thank you for the changes and updating everything. I just added some minor corrections.
> 
> The docs build has been successfully triggered now, hasn't it?

Yes :)

**Pull request merged**